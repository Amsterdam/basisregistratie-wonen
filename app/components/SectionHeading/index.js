import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const H3 = styled.h3`
  margin-top: 30px;
  ${({ marginCollapse }) =>
    marginCollapse &&
    css`
      margin-top: 0;
    `}

  @media print {
    margin-top: 0;
  }
`;

const SectionHeading = ({ children, ...rest }) => <H3 {...rest}>{children}</H3>;

SectionHeading.defaultProps = {
  marginCollapse: false,
};

SectionHeading.propTypes = {
  children: PropTypes.node.isRequired,
  marginCollapse: PropTypes.bool,
};

export default SectionHeading;
