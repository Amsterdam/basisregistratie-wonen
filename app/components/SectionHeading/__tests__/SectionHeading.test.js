import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-styled-components';

import SectionHeading from '..';

describe('SectionHeading', () => {
  afterEach(cleanup);

  it('should match the snapshot', () => {
    const { container, rerender } = render(
      <SectionHeading marginCollapse>Here be the section heading content</SectionHeading>,
    );
    expect(container.firstChild).toHaveStyleRule('margin-top', '0');
    expect(container.firstChild).toMatchSnapshot();

    rerender(<SectionHeading>Text</SectionHeading>);
    expect(container.firstChild).toHaveStyleRule('margin-top', '0', {
      media: 'print',
    });
    expect(container.firstChild).not.toHaveStyleRule('margin-top', '0');
  });
});
