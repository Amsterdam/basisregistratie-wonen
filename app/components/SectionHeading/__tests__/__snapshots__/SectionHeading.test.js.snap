// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeading should match the snapshot 1`] = `
.c0 {
  margin-top: 30px;
  margin-top: 0;
}

@media print {
  .c0 {
    margin-top: 0;
  }
}

<h3
  class="c0"
>
  Here be the section heading content
</h3>
`;
